WeatherMonitoringAndReportingService/DataSourceProcessor/JSONFileProcessor.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using WeatherMonitoringAndReportingService.AppSettings;
+using WeatherMonitoringAndReportingService.Config;
+
+namespace WeatherMonitoringAndReportingService.DataSourceProcessor;
+
+public class JSONFileProcessor : IDataSourceProcessor
+{
+    public void Add(string name, WeatherConfigurationModel data, string path)
+    {
+        path ??= AppSettingsInitializer.AppSettingsInstance().ConfigFilePath;
+
+        // 1. Read file
+        var currentData = ReadFile(path);
+
+        // 2. Add 
+        currentData.Add(name, data);
+
+        // Serialize Dictionary
+        var serializedData = Serialize(currentData);
+
+        // Write file
+        WriteFile(serializedData, path);
+    }
+
+    public Dictionary<string, WeatherConfigurationModel> ReadFile(string? path)
+    {
+        string jsonString = File.ReadAllText(path ?? AppSettingsInitializer.AppSettingsInstance().ConfigFilePath);
+        var botsSettings = JsonSerializer.Deserialize<Dictionary<string, WeatherConfigurationModel>>(jsonString);
+
+        return botsSettings;
```

Comment:
Might want to handle the possible null from the `JsonSerializer.Deserialize` method. Either throw an exception or use the `!` operator. Our application will never expect the file to be missing but it would be better if we fail if it isn't there or handle it appropriately.